src/lib/__tests__/InstantSearch-test.js

```diff
@@ -252,6 +252,50 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/instantsear
 See documentation: https://www.algolia.com/doc/api-reference/widgets/instantsearch/js/"
 `);
   });
+
+  it('warns dev with EXPERIMENTAL_use', () => {
+    const searchClient = createSearchClient({
+      addAlgoliaAgent: jest.fn(),
+    });
+
+    // eslint-disable-next-line no-new
+    const search = new InstantSearch({
+      indexName: 'indexName',
+      searchClient,
+    });
+
+    const middleware = () => ({
+      onStateChange: () => {},
+      subscribe: () => {},
+      unsubscribe: () => {},
+    });
+
+    expect(() => {
+      search.EXPERIMENTAL_use(middleware);
+    }).toWarnDev();
```

Comment:
We better assert the actual message because it has happened that other warnings were printed when calling a lifecycle method.

Reply:
https://github.com/algolia/instantsearch.js/pull/4450/commits/5cd0d7983d94bc8442b566aec3ca0dcd27371816